src/components/SharedComponents/TaxonResult.tsx

```diff
@@ -192,51 +230,52 @@ const TaxonResult = ( {
             </View>
           )}
         </View>
-      </Pressable>
-      <View className="flex-row items-center">
-        { !hideInfoButton && (
-          <INatIconButton
-            icon="info-circle-outline"
-            size={22}
-            onPress={( ) => {
-              if ( typeof ( onPressInfo ) === "function" ) {
-                onPressInfo( usableTaxon );
-                return;
-              }
-              navToTaxonDetails( );
-            }}
-            color={String(
-              clearBackground
-                ? colors.white
-                : colors.darkGray
-            )}
-            accessibilityLabel={t( "More-info" )}
-            accessibilityHint={t( "Navigates-to-taxon-details" )}
-          />
-        )}
-        { showCheckmark
-          && renderCheckmark()}
-        { showEditButton
+      </TaxonResultMain>
+      { !unpressable && (
```

Comment:
An alternative approach to consider: using `unpressable` with boolean operations to modify the values of `hideInfoButton` and `showCheckmark` and `showEditButton` either inline below or when unpacking `TaxonResultProps` would avoid the nesting introduced here. This might help future readers to understand which specific elements are being rendered or hidden, and why.

Reply:
I hear you, but personally when I'm reading I like to see variables close to their effects. If `hideInfoButton` hides an info button, I'd like to see the boolean get checked close to where the info button is rendered so I'm not jumping up and down in the file. IMO that's worth a little extra indentation. If I'm misinterpreting, could you provide an example?

Reply:
Agree about variables being close to their effects. I also want to avoid adding complex boolean logic at lower levels (e.g. I would not support L236 changing from ` { !hideInfoButton && (` to ` { !hideInfoButton && !unpressable && (`)
What I was trying to suggest was that somewhere prior to this view being rendered, we'd have applied the effects of the override to the other boolean values like this: `hideInfoButton ||= unpressable; showCheckmark &&= !unpressable;`
That's marginally readable, but it's as far as I thought would be reasonable to suggest without asking for a complete refactor of the file.